### PR TITLE
api: fix missing individual resource routes and minor refactor

### PR DESCRIPTION
**Pull request by andrewbonney**

### CODE REVIEW

File: nmosnode/api.py [hunk 1]
```diff
@@ -54,19 +55,29 @@ def versionroot(self, api_version):
...
+    def resource_id(self, api_version, resource_type, resource_id):
         if api_version not in NODE_APIVERSIONS:
             abort(404)
-        receivers = self.registry.list_resource("receiver", api_version=api_version)
-        if receiver in receivers:
-            return receivers[receiver]
+        if resource_type not in RESOURCE_TYPES:
+            abort(404)
+        resources = self.registry.list_resource(resource_type.rstrip("s"), api_version=api_version)
+        if resource_id in resources:
```
Comment (simonrankine) [inline nmosnode/api.py:75]: Doesn't need changing, but I'd normally expected to see this done with a try/catch block in Python

Comment (andrewbonney) [reply nmosnode/api.py:75]: Yes, I have a bit of a habit of not being fully Pythonic...

File: nmosnode/api.py [hunk 2]
```diff
@@ -54,19 +55,29 @@ def versionroot(self, api_version):
...
-    @resource_route('/'+NODE_APINAMESPACE+'/'+NODE_APINAME+"/<api_version>/senders/")
-    def senders(self, api_version):
+    @route('/'+NODE_APINAMESPACE+'/'+NODE_APINAME+"/<api_version>/self/")
+    def selfresource(self, api_version):
         if api_version not in NODE_APIVERSIONS:
             abort(404)
-        return self.registry.list_resource("sender", api_version=api_version).values()
+        return self.registry.list_self(api_version=api_version)
+
+    @resource_route('/'+NODE_APINAMESPACE+'/'+NODE_APINAME+"/<api_version>/<resource_type>/")
```
Comment (simonrankine) [inline nmosnode/api.py:58]: Flask match priority seems to be complicated, but I'm kinda worried this route also matches the /self/ route defined above - meaning that there is the potential for flask to decide to use this method for /self and end up with this kind of issue: https://stackoverflow.com/questions/17868916/flask-route-pattern-matching-order

**Review by simonrankine — APPROVED** [review]

File: nmosnode/api.py (hunk 1 above)
Comment (simonrankine) [reply nmosnode/api.py:75]: It's fine, I catch myself doing this sort of thing too but worth bearing in mind ;-)

File: nmosnode/api.py (hunk 2 above)
Comment (andrewbonney) [reply nmosnode/api.py:58]: I wonder if they've improved the logic since then. I could easily do this via a single method if required.

**Pull request merged**